service/src/LevelZero.cpp

```diff
@@ -628,46 +620,15 @@ namespace geopm
                "ZE_RESULT_ERROR_HANDLE_OBJECT_IN_USE"},
               {ZE_RESULT_ERROR_INVALID_NULL_POINTER,
                "ZE_RESULT_ERROR_INVALID_NULL_POINTER"},
-              {ZE_RESULT_ERROR_INVALID_SIZE,
-               "ZE_RESULT_ERROR_INVALID_SIZE"},
-              {ZE_RESULT_ERROR_UNSUPPORTED_SIZE,
-               "ZE_RESULT_ERROR_UNSUPPORTED_SIZE"},
-              {ZE_RESULT_ERROR_UNSUPPORTED_ALIGNMENT,
-               "ZE_RESULT_ERROR_UNSUPPORTED_ALIGNMENT"},
-              {ZE_RESULT_ERROR_INVALID_SYNCHRONIZATION_OBJECT,
-               "ZE_RESULT_ERROR_INVALID_SYNCHRONIZATION_OBJECT"},
-              {ZE_RESULT_ERROR_INVALID_ENUMERATION,
-               "ZE_RESULT_ERROR_INVALID_ENUMERATION"},
-              {ZE_RESULT_ERROR_UNSUPPORTED_ENUMERATION,
-               "ZE_RESULT_ERROR_UNSUPPORTED_ENUMERATION"},
-              {ZE_RESULT_ERROR_UNSUPPORTED_IMAGE_FORMAT,
-               "ZE_RESULT_ERROR_UNSUPPORTED_IMAGE_FORMAT"},
-              {ZE_RESULT_ERROR_INVALID_NATIVE_BINARY,
-               "ZE_RESULT_ERROR_INVALID_NATIVE_BINARY"},
-              {ZE_RESULT_ERROR_INVALID_GLOBAL_NAME,
-               "ZE_RESULT_ERROR_INVALID_GLOBAL_NAME"},
-              {ZE_RESULT_ERROR_INVALID_KERNEL_NAME,
-               "ZE_RESULT_ERROR_INVALID_KERNEL_NAME"},
-              {ZE_RESULT_ERROR_INVALID_FUNCTION_NAME,
-               "ZE_RESULT_ERROR_INVALID_FUNCTION_NAME"},
-              {ZE_RESULT_ERROR_INVALID_GROUP_SIZE_DIMENSION,
-               "ZE_RESULT_ERROR_INVALID_GROUP_SIZE_DIMENSION"},
-              {ZE_RESULT_ERROR_INVALID_GLOBAL_WIDTH_DIMENSION,
-               "ZE_RESULT_ERROR_INVALID_GLOBAL_WIDTH_DIMENSION"},
-              {ZE_RESULT_ERROR_INVALID_KERNEL_ARGUMENT_INDEX,
-               "ZE_RESULT_ERROR_INVALID_KERNEL_ARGUMENT_INDEX"},
-              {ZE_RESULT_ERROR_INVALID_KERNEL_ARGUMENT_SIZE,
-               "ZE_RESULT_ERROR_INVALID_KERNEL_ARGUMENT_SIZE"},
-              {ZE_RESULT_ERROR_INVALID_KERNEL_ATTRIBUTE_VALUE,
-               "ZE_RESULT_ERROR_INVALID_KERNEL_ATTRIBUTE_VALUE"},
-              {ZE_RESULT_ERROR_INVALID_COMMAND_LIST_TYPE,
-               "ZE_RESULT_ERROR_INVALID_COMMAND_LIST_TYPE"},
-              {ZE_RESULT_ERROR_OVERLAPPING_REGIONS,
-               "ZE_RESULT_ERROR_OVERLAPPING_REGIONS"},
               {ZE_RESULT_ERROR_UNKNOWN,
                "ZE_RESULT_ERROR_UNKNOWN"},
             };
-        std::string error_string = error_mapping.at(ze_result);
+
+        std::string error_string = std::to_string(ze_result);
+        if (error_mapping.find(ze_result) != error_mapping.end()) {
+            error_string = error_mapping.at(ze_result);
+        }
```

Comment:
Very minor suggestion, not to stop the review:

```suggestion
        const auto it = error_mapping.find(ze_result);
        if (it != error_mapping.end()) {
            error_string = it->second;
        }
```

Reply:
Seems reasonable to me!